File: mebelonline.client/src/models/categoryDetailsModel.ts
export type CategoryRefModel = {
  id: number;
  name: string;
};

export type CategoryParentModel = {
  id: number;
  name: string;
  parent?: CategoryParentModel | null;
};

export type CategoryDetailsModel = {
  id: number;
  name: string;
  imageUrl?: string | null;
  parent?: CategoryParentModel | null;
  children?: CategoryRefModel[] | null;
};

File: MebelOnline.Core/Models/Categories/CategoryDetailsModel.cs
namespace MebelOnline.Core.Models.Categories
{
    public class CategoryDetailsModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? ImageUrl { get; set; }
        public CategoryParentModel? Parent { get; set; }
        public IList<CategorySummaryModel> Children { get; set; } = new List<CategorySummaryModel>();
    }
}
